macros/models/_sector/dex/enrich_dex_trades.sql

```diff
@@ -0,0 +1,70 @@
+{% macro enrich_dex_trades(
+    base_trades = null,
+    tokens_erc20_model=null,
+    prices_model=null
+    )
+%}
+
+WITH prices AS (
+    SELECT
+        blockchain,
+        contract_address,
+        minute,
+        price
+    FROM {{ prices_model }}
+    {% if is_incremental() %}
+    WHERE {{ incremental_predicate('minute') }}
+    {% endif %}
+),
+
+enrichments AS (
+    SELECT
+        base.blockchain,
+        base.project,
+        base.version,
+        base.block_month,
+        base.block_date,
+        base.block_time,
+        base.block_number,
+        erc20_bought.symbol AS token_bought_symbol,
+        erc20_sold.symbol AS token_sold_symbol,
+        case
+            when lower(erc20_bought.symbol) > lower(erc20_sold.symbol) then concat(erc20_sold.symbol, '-', erc20_bought.symbol)
+            else concat(erc20_bought.symbol, '-', erc20_sold.symbol)
+            end AS token_pair,
+        base.token_bought_amount_raw / power(10, erc20_bought.decimals) AS token_bought_amount,
+        base.token_sold_amount_raw / power(10, erc20_sold.decimals) AS token_sold_amount,
+        base.token_bought_amount_raw,
+        base.token_sold_amount_raw,
+        coalesce(
+                base.token_bought_amount_raw / power(10, erc20_bought.decimals) * p_bought.price,
+                base.token_sold_amount_raw / power(10, erc20_sold.decimals) * p_sold.price
+            ) AS amount_usd,
+        base.token_bought_address,
+        base.token_sold_address,
+        coalesce(base.taker, base.tx_from) AS taker,
+        base.maker,
+        base.project_contract_address,
+        base.tx_hash,
+        base.tx_from,
+        base.tx_to,
+        base.evt_index
```

Comment:
some of the projects just don't emit events. i think it'd be great to add `trace_address` or `call_trace_address` as in current **dex_aggregator.trades** implementation. or just merge it in one column, but it should be named clearly.

Reply:
we just removed `trace_address` in the migration to dunesql, since not a single project within the `dex.trades` lineage provided a value for this column. they all populated events instead. i'd be curious to see examples of dex projects in the existing lineage that would use this field?

Reply:
i can't find examples :)

thanks for sharing the context!